src/iio_wrapper.cpp

```diff
@@ -120,19 +177,36 @@ IIOWrapper::IIOWrapper()
 IIOWrapper::~IIOWrapper()
```

Comment:
The m_device_buffer seems left un-destroyed.
The m_channels doesn't seem to be freed anywhere.